### PR TITLE
Feature/jacobayu/admin routes

### LINKED ISSUE #27: Admin schema and routes

### PR DESCRIPTION
**Pull request by jacobayu**
### Administrative Info
[GitHub Issue #]
Make sure your branch name conforms to: `<feature/staging/hotfix/...>/[username]/[3-4 word description separated by dashes]`. Otherwise, please rename your branch and create a new PR.

### Changes
What changes did you make?
- Added admins scheme and routes

### Testing
How did you confirm your changes worked? 
- Used postman to make sure update, create, and login routes all worked correctly

### CODE REVIEW

**Review by parth4apple — APPROVED** [review]
LGTM. 
It looks like you modified your package-lock.json though, which might cause a merge conflict. You can address this by merging the main branch into your branch, deleting the package-loc.json file, and running npm install after. 
"git merge main" is the command to merge main into your branch

**Review by dnanjunda — CHANGES_REQUESTED** [review]
Looks clean Jacob! There's a couple bugs I noticed and also a couple things I realized need to be added:

1. In your validators variable, make sure to check if phone is a mobile number, if email is in email address format, and if photoURL is an actual url.
2. For all routes, when there's an error (when the user couldn't be validated, when the profile couldn't be edited, etc.), send a res status 500 message instead of throwing an error.
3. We actually need two more routes for admins. One route to get all admin profiles and another to get a specific admin using an ID that will be passed in. These routes will be used for the profile page. I'll add details for these routes in the Project Details doc as well.

Other notes (just for bookkeeping):
1. The login route will probably change (once authentication is setup).
2. The schedule field will probably be altered depending on client's needs.

File: models/admin.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      unique: true,
+    },
+    phone: {
+      type: String,
+      required: true,
+    },
+    password: {
+      type: String,
+      required: true,
+    },
```
Comment (petabite) [inline models/admin.js:31]: Since we are using Firebase for authentication now, I don't think we need to save passwords.

**Review by dnanjunda — CHANGES_REQUESTED** [review]
Almost there Jacob! Couple minor things:

1. Can you send the error message through res status 500 for the "create an admin" post route, like you did for the other routes?
2. It looks like the phone number field still allows less than 10 digits. You might have to specify the country in isMobilePhone().
3. For the schedule field, make sure to check that the object isn't empty. Otherwise, you can just pass in an empty object and that field won't exist.

**Review by dnanjunda — APPROVED** [review]
LGTM

**Pull request merged**